kentikapi/internal/api_payloads/cloud_export.go

```diff
@@ -251,3 +258,75 @@ func boolProtoPtrToBoolPtr(v *wrapperspb.BoolValue) *bool {
 	}
 	return pointer.ToBool(v.GetValue())
 }
+
+// ValidateCECreateRequest checks if CloudExport create request contains all required fields.
+func ValidateCECreateRequest(ce *models.CloudExport) error {
+	if ce == nil {
+		return fmt.Errorf("cloud export object is nil")
+	}
+	if ce.Name == "" {
+		return ceFieldError("Name")
+	}
+	if ce.PlanID == "" {
+		return ceFieldError("PlanID")
+	}
+	return validateCEProvider(ce)
+}
+
+// ValidateCEUpdateRequest checks if CloudExport update request contains all required fields.
+func ValidateCEUpdateRequest(ce *models.CloudExport) error {
+	if ce == nil {
+		return fmt.Errorf("cloud export object is nil")
+	}
+	if ce.ID == "" {
+		return ceFieldError("ID")
+	}
+	if ce.Name == "" {
+		return ceFieldError("Name")
+	}
+	if ce.PlanID == "" {
+		return ceFieldError("PlanID")
+	}
+	return validateCEProvider(ce)
+}
+
+//nolint:gocyclo
+func validateCEProvider(ce *models.CloudExport) error {
+	switch ce.CloudProvider {
+	case "":
+		return ceFieldError("CloudProvider")
+	case awsProvider:
+		if ce.GetAWSProperties().Bucket == "" {
```

Comment:
```suggestion
                if ce.GetAWSProperties() == nil {return ceFieldError("Properties") }
		if ce.GetAWSProperties().Bucket == "" {
```
GetAWSProperties() might return nil pointer and nil pointer dereference causes panic. We need to do nil check to avoid that.

Reply:
This comment also applies to other providers, e.g. Azure

Reply:
Done